finetuner/tuner/onnx.py

```diff
@@ -0,0 +1,206 @@
+from typing import Any, List, Tuple, Union
+
+import numpy as np
+import onnxruntime
+
+from finetuner.helper import AnyDNN, get_framework
+
+
+def to_onnx(
+    embed_model: AnyDNN,
+    path: str,
+    input_shape: Union[Tuple[int], List[int]],
+    opset_version: int = 11,
+) -> None:
+    """Func that converts a given model in paddle, torch or keras,
+    and converts it to the ONNX format
+    :param embed_model: Model to be converted and stored in ONNX
+    :param path: Path to store ONNX model to
+    :param input_shape: Input shape of embedding model
+    :param opset_version: ONNX opset version in which to register
+    """
+    if isinstance(input_shape, tuple):
+        input_shape = list(input_shape)
+
+    if not path.endswith(".onnx"):
+        raise ValueError(f"The `path` needs to end with `.onnx`, but was: {path}")
+
+    def _parse_and_apply_to_onnx_func(framework_name: str):
+        """Helper func to get _to_onnx_xyz func from framework name"""
+        return {
+            'torch': _to_onnx_torch,
+            'keras': _to_onnx_keras,
+            'paddle': _to_onnx_paddle,
+        }[framework_name](embed_model, path, input_shape, opset_version)
+
+    fm = get_framework(embed_model)
+
+    # Get framework-specific func to register model in ONNX
+    _parse_and_apply_to_onnx_func(fm)
+
+    _check_onnx_model(path)
+
+
+def _check_onnx_model(path: str) -> None:
+    """Check an ONNX model
+    :param path: Path to ONNX model
+    """
+    import onnx
+
+    model = onnx.load(path)
+    onnx.checker.check_model(model)
+
+
+def _to_onnx_torch(
+    embed_model: AnyDNN,
+    path: str,
+    input_shape: Tuple[int, ...],
+    opset_version: int = 11,
+    batch_size: int = 16,
+) -> None:
+    """Convert a PyTorch embedding model to the ONNX format
+    :param embed_model: Embedding model to register in ONNX
+    :param path: Patch where to register ONNX model to
+    :param input_shape: Embedding model input shape
+    :param batch_size: The batch size during export
+    :param opset_version: ONNX opset version in which to register
+    """
+
+    import torch
+
+    embed_model.eval()
+
+    x = torch.randn([batch_size] + input_shape, requires_grad=True, dtype=torch.float32)
```

Comment:
why we stick with `float32` as dtype? does it work for different kind of models? we have `input_dtype` as `float32` as default in tailor, i think we need to have some flexibility here. see [this](https://github.com/jina-ai/finetuner/blob/main/finetuner/tailor/base.py#L13) link

Reply:
But in tailor you don't even use that property, or am i wrong?

Reply:
we use, actually for most of the models `float32` works, lstm seems does not (with an `Embedding` layer).

Reply:
I am curious, why is this even needed to export a model.

Reply:
It is weird that to export a model I need to pass a batch_size as input

Reply:
Creating an example input is needed for pytorch to trace the model for exporting - but we could have a single fixed value for batch size, since it is set as dynamic anyway

Reply:
at least yes, u shouild have a single squeezed input

Reply:
But let's not expose `batch-size`

Reply:
I've added the param